Bottom: b0ee408136569761d1b16490abed515ba5c2204b
Top:    38d8efa5c4871ecc7d53042245a9045be3389f65
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2012-11-14 21:59:49 +0000

Remove HID->set_draw_xor() API

This is only used for drawing the reference mark and objects attached
to the cursor. Rather than have the core setup the XOR drawing mode,
let the GUI set what it wants.


---

diff --git a/src/crosshair.c b/src/crosshair.c
index f0526b8..f6498f7 100644
--- a/src/crosshair.c
+++ b/src/crosshair.c
@@ -1174,7 +1174,6 @@ InitCrosshair (void)
   Crosshair.GC = gui->make_gc ();
 
   gui->set_color (Crosshair.GC, Settings.CrosshairColor);
-  gui->set_draw_xor (Crosshair.GC, 1);
   gui->set_line_cap (Crosshair.GC, Trace_Cap);
   gui->set_line_width (Crosshair.GC, 1);
 
diff --git a/src/hid.h b/src/hid.h
index 28daf1d..36f80ed 100644
--- a/src/hid.h
+++ b/src/hid.h
@@ -357,7 +357,6 @@ typedef enum
        line style is helpful.  */
     void (*set_line_cap) (hidGC gc_, EndCapStyle style_);
     void (*set_line_width) (hidGC gc_, Coord width_);
-    void (*set_draw_xor) (hidGC gc_, int xor_);
     /* Blends 20% or so color with 80% background.  Only used for
        assembly drawings so far. */
     void (*set_draw_faded) (hidGC gc_, int faded_);
diff --git a/src/hid/batch/batch.c b/src/hid/batch/batch.c
index f49bc19..824cbaa 100644
--- a/src/hid/batch/batch.c
+++ b/src/hid/batch/batch.c
@@ -200,11 +200,6 @@ batch_set_line_width (hidGC gc, Coord width)
 }
 
 static void
-batch_set_draw_xor (hidGC gc, int xor_set)
-{
-}
-
-static void
 batch_draw_line (hidGC gc, Coord x1, Coord y1, Coord x2, Coord y2)
 {
 }
@@ -351,7 +346,6 @@ hid_batch_init ()
   batch_hid.set_color             = batch_set_color;
   batch_hid.set_line_cap          = batch_set_line_cap;
   batch_hid.set_line_width        = batch_set_line_width;
-  batch_hid.set_draw_xor          = batch_set_draw_xor;
   batch_hid.draw_line             = batch_draw_line;
   batch_hid.draw_arc              = batch_draw_arc;
   batch_hid.draw_rect             = batch_draw_rect;
diff --git a/src/hid/common/extents.c b/src/hid/common/extents.c
index d970c77..0df42f0 100644
--- a/src/hid/common/extents.c
+++ b/src/hid/common/extents.c
@@ -91,11 +91,6 @@ extents_set_line_width (hidGC gc, Coord width)
   gc->width = width;
 }
 
-static void
-extents_set_draw_xor (hidGC gc, int xor_)
-{
-}
-
 #define PEX(x,w) if (box.X1 > (x)-(w)) box.X1 = (x)-(w); \
 	if (box.X2 < (x)+(w)) box.X2 = (x)+(w)
 #define PEY(y,w) if (box.Y1 > (y)-(w)) box.Y1 = (y)-(w); \
@@ -181,7 +176,6 @@ hid_extents_init (void)
   extents_hid.set_color           = extents_set_color;
   extents_hid.set_line_cap        = extents_set_line_cap;
   extents_hid.set_line_width      = extents_set_line_width;
-  extents_hid.set_draw_xor        = extents_set_draw_xor;
   extents_hid.draw_line           = extents_draw_line;
   extents_hid.draw_arc            = extents_draw_arc;
   extents_hid.draw_rect           = extents_draw_rect;
diff --git a/src/hid/common/hidnogui.c b/src/hid/common/hidnogui.c
index d772e18..5685d60 100644
--- a/src/hid/common/hidnogui.c
+++ b/src/hid/common/hidnogui.c
@@ -105,12 +105,6 @@ nogui_set_line_width (hidGC gc, Coord width)
 }
 
 static void
-nogui_set_draw_xor (hidGC gc, int xor_)
-{
-  CRASH;
-}
-
-static void
 nogui_set_draw_faded (hidGC gc, int faded)
 {
 }
@@ -460,7 +454,6 @@ common_nogui_init (HID *hid)
   hid->set_color =            nogui_set_color;
   hid->set_line_cap =         nogui_set_line_cap;
   hid->set_line_width =       nogui_set_line_width;
-  hid->set_draw_xor =         nogui_set_draw_xor;
   hid->set_draw_faded =       nogui_set_draw_faded;
   hid->draw_line =            nogui_draw_line;
   hid->draw_arc =             nogui_draw_arc;
diff --git a/src/hid/gcode/gcode.c b/src/hid/gcode/gcode.c
index 33c4a00..a2595c2 100644
--- a/src/hid/gcode/gcode.c
+++ b/src/hid/gcode/gcode.c
@@ -1230,12 +1230,6 @@ gcode_set_line_width (hidGC gc, Coord width)
 }
 
 static void
-gcode_set_draw_xor (hidGC gc, int xor_)
-{
-  ;
-}
-
-static void
 gcode_set_draw_faded (hidGC gc, int faded)
 {
 }
@@ -1594,7 +1588,6 @@ hid_gcode_init ()
   gcode_hid.set_color           = gcode_set_color;
   gcode_hid.set_line_cap        = gcode_set_line_cap;
   gcode_hid.set_line_width      = gcode_set_line_width;
-  gcode_hid.set_draw_xor        = gcode_set_draw_xor;
   gcode_hid.set_draw_faded      = gcode_set_draw_faded;
   gcode_hid.draw_line           = gcode_draw_line;
   gcode_hid.draw_arc            = gcode_draw_arc;
diff --git a/src/hid/gerber/gerber.c b/src/hid/gerber/gerber.c
index 79c0447..a41ba26 100644
--- a/src/hid/gerber/gerber.c
+++ b/src/hid/gerber/gerber.c
@@ -52,7 +52,6 @@ static void gerber_use_mask (int use_it);
 static void gerber_set_color (hidGC gc, const char *name);
 static void gerber_set_line_cap (hidGC gc, EndCapStyle style);
 static void gerber_set_line_width (hidGC gc, Coord width);
-static void gerber_set_draw_xor (hidGC gc, int _xor);
 static void gerber_draw_line (hidGC gc, Coord x1, Coord y1, Coord x2, Coord y2);
 static void gerber_draw_arc (hidGC gc, Coord cx, Coord cy, Coord width, Coord height, Angle start_angle, Angle delta_angle);
 static void gerber_draw_rect (hidGC gc, Coord x1, Coord y1, Coord x2, Coord y2);
@@ -932,12 +931,6 @@ gerber_set_line_width (hidGC gc, Coord width)
 }
 
 static void
-gerber_set_draw_xor (hidGC gc, int xor_)
-{
-  ;
-}
-
-static void
 use_gc (hidGC gc, int radius)
 {
   if (radius)
@@ -1284,7 +1277,6 @@ hid_gerber_init ()
   gerber_hid.set_color           = gerber_set_color;
   gerber_hid.set_line_cap        = gerber_set_line_cap;
   gerber_hid.set_line_width      = gerber_set_line_width;
-  gerber_hid.set_draw_xor        = gerber_set_draw_xor;
   gerber_hid.draw_line           = gerber_draw_line;
   gerber_hid.draw_arc            = gerber_draw_arc;
   gerber_hid.draw_rect           = gerber_draw_rect;
diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index a1c6e5b..529a571 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -484,16 +484,6 @@ ghid_set_line_width (hidGC gc, Coord width)
   gc->width = width;
 }
 
-
-void
-ghid_set_draw_xor (hidGC gc, int xor)
-{
-  /* NOT IMPLEMENTED */
-
-  /* Only presently called when setting up a crosshair GC.
-   * We manage our own drawing model for that anyway. */
-}
-
 void
 ghid_set_draw_faded (hidGC gc, int faded)
 {
diff --git a/src/hid/gtk/gtkhid-main.c b/src/hid/gtk/gtkhid-main.c
index bf1885a..05fdfd2 100644
--- a/src/hid/gtk/gtkhid-main.c
+++ b/src/hid/gtk/gtkhid-main.c
@@ -2150,7 +2150,6 @@ hid_gtk_init ()
   ghid_hid.set_color                = ghid_set_color;
   ghid_hid.set_line_cap             = ghid_set_line_cap;
   ghid_hid.set_line_width           = ghid_set_line_width;
-  ghid_hid.set_draw_xor             = ghid_set_draw_xor;
   ghid_hid.draw_line                = ghid_draw_line;
   ghid_hid.draw_arc                 = ghid_draw_arc;
   ghid_hid.draw_rect                = ghid_draw_rect;
diff --git a/src/hid/gtk/gui.h b/src/hid/gtk/gui.h
index 77fa4d4..e6a95e8 100644
--- a/src/hid/gtk/gui.h
+++ b/src/hid/gtk/gui.h
@@ -476,7 +476,6 @@ void ghid_use_mask (int use_it);
 void ghid_set_color (hidGC gc, const char *name);
 void ghid_set_line_cap (hidGC gc, EndCapStyle style);
 void ghid_set_line_width (hidGC gc, Coord width);
-void ghid_set_draw_xor (hidGC gc, int _xor);
 void ghid_draw_line (hidGC gc, Coord x1, Coord y1, Coord x2, Coord y2);
 void ghid_draw_arc (hidGC gc, Coord cx, Coord cy, Coord xradius, Coord yradius,
                     Angle start_angle, Angle delta_angle);
diff --git a/src/hid/nelma/nelma.c b/src/hid/nelma/nelma.c
index 048f476..402e0d0 100644
--- a/src/hid/nelma/nelma.c
+++ b/src/hid/nelma/nelma.c
@@ -819,12 +819,6 @@ nelma_set_line_width(hidGC gc, Coord width)
 }
 
 static void
-nelma_set_draw_xor(hidGC gc, int xor_)
-{
-	;
-}
-
-static void
 nelma_set_draw_faded(hidGC gc, int faded)
 {
 }
@@ -1056,7 +1050,6 @@ hid_nelma_init()
   nelma_hid.set_color           = nelma_set_color;
   nelma_hid.set_line_cap        = nelma_set_line_cap;
   nelma_hid.set_line_width      = nelma_set_line_width;
-  nelma_hid.set_draw_xor        = nelma_set_draw_xor;
   nelma_hid.set_draw_faded      = nelma_set_draw_faded;
   nelma_hid.draw_line           = nelma_draw_line;
   nelma_hid.draw_arc            = nelma_draw_arc;
diff --git a/src/hid/png/png.c b/src/hid/png/png.c
index c7c0efd..5bfa5f6 100644
--- a/src/hid/png/png.c
+++ b/src/hid/png/png.c
@@ -1463,12 +1463,6 @@ png_set_line_width (hidGC gc, Coord width)
 }
 
 static void
-png_set_draw_xor (hidGC gc, int xor_)
-{
-  ;
-}
-
-static void
 use_gc (hidGC gc)
 {
   int need_brush = 0;
@@ -1822,7 +1816,6 @@ hid_png_init ()
   png_hid.set_color           = png_set_color;
   png_hid.set_line_cap        = png_set_line_cap;
   png_hid.set_line_width      = png_set_line_width;
-  png_hid.set_draw_xor        = png_set_draw_xor;
   png_hid.draw_line           = png_draw_line;
   png_hid.draw_arc            = png_draw_arc;
   png_hid.draw_rect           = png_draw_rect;
diff --git a/src/hid/ps/eps.c b/src/hid/ps/eps.c
index 0e90c7b..b2fda62 100644
--- a/src/hid/ps/eps.c
+++ b/src/hid/ps/eps.c
@@ -39,7 +39,6 @@ static void eps_use_mask (int use_it);
 static void eps_set_color (hidGC gc, const char *name);
 static void eps_set_line_cap (hidGC gc, EndCapStyle style);
 static void eps_set_line_width (hidGC gc, Coord width);
-static void eps_set_draw_xor (hidGC gc, int _xor);
 static void eps_draw_rect (hidGC gc, Coord x1, Coord y1, Coord x2, Coord y2);
 static void eps_draw_line (hidGC gc, Coord x1, Coord y1, Coord x2, Coord y2);
 static void eps_draw_arc (hidGC gc, Coord cx, Coord cy, Coord width, Coord height, Angle start_angle, Angle delta_angle);
@@ -517,12 +516,6 @@ eps_set_line_width (hidGC gc, Coord width)
 }
 
 static void
-eps_set_draw_xor (hidGC gc, int xor_)
-{
-  ;
-}
-
-static void
 use_gc (hidGC gc)
 {
   if (linewidth != gc->width)
@@ -685,7 +678,6 @@ hid_eps_init ()
   eps_hid.set_color           = eps_set_color;
   eps_hid.set_line_cap        = eps_set_line_cap;
   eps_hid.set_line_width      = eps_set_line_width;
-  eps_hid.set_draw_xor        = eps_set_draw_xor;
   eps_hid.draw_line           = eps_draw_line;
   eps_hid.draw_arc            = eps_draw_arc;
   eps_hid.draw_rect           = eps_draw_rect;
diff --git a/src/hid/ps/ps.c b/src/hid/ps/ps.c
index 5587347..a43a861 100644
--- a/src/hid/ps/ps.c
+++ b/src/hid/ps/ps.c
@@ -1067,12 +1067,6 @@ ps_set_line_width (hidGC gc, Coord width)
 }
 
 static void
-ps_set_draw_xor (hidGC gc, int xor_)
-{
-  ;
-}
-
-static void
 ps_set_draw_faded (hidGC gc, int faded)
 {
   gc->faded = faded;
@@ -1501,7 +1495,6 @@ void ps_ps_init (HID *hid)
   hid->set_color          = ps_set_color;
   hid->set_line_cap       = ps_set_line_cap;
   hid->set_line_width     = ps_set_line_width;
-  hid->set_draw_xor       = ps_set_draw_xor;
   hid->set_draw_faded     = ps_set_draw_faded;
   hid->draw_line          = ps_draw_line;
   hid->draw_arc           = ps_draw_arc;
